src/java.base/share/classes/java/lang/ref/Reference.java

```diff
@@ -342,6 +342,20 @@ public T get() {
         return this.referent;
     }
 
+    /**
+     * Load referent with strong semantics. Treating the referent
+     * as strong referent is ok when the Reference is inactive,
+     * because then the referent is switched to strong semantics
+     * anyway.
+     *
+     * This is only used from Finalizer to bypass the intrinsic,
+     * which might return a null referent, even though it is not
+     * null, and would subsequently not finalize the referent/finalizee.
+     */
+    T getInactive() {
+        return this.referent;
```

Comment:
It would be good to add `assert this instanceof FinalRefernce` to make this assertion clear.

Reply:
Right. And maybe also assert that the Reference is indeed inactive. I'll open a new issue for that (I already integrated this one, sorry I kinda jumped the gun a little.)